docs/technical-design/endpoints.md

```diff
@@ -0,0 +1,213 @@
+
+```typescript
+dataExport: dataExportResolver(store),
+```
+
+### Generating code
+
+Once we have our backend code in place, we need to run a couple of commands to generate glue types and methods.
+
+_**On the command line, navigate to `/services/app-graphql`.**_  
+_**Run `yarn gqlgen`**_
+
+_**Again on the command line, navigate to `/services/app-api`**_  
+_**Run `yarn prisma generate`**_
+
+### Getting data on the front end
+
+One result of the `generate` commands is to create a hook that we can use on the front end to call our new resolver. The name of the hook will be of the form _useWhatYourResolverIsCalledQuery_ (it will end with "Mutation" if that's what you've created). You can pull it in and get data.
+
+```typescript
+import { useDataExportQuery } from '../../gen/gqlClient'
+const { data, loading, error } = useDataExportQuery()
+```
+
+## Lambda
+
+### Serverless configuration
+
+When we create a handler, we're telling serverless to create a new Lambda function for us in AWS. If we configure it correctly, that function will have access to things like our postgres database, which itself is hosted on AWS. Here's one existing handler configuration, found in `/services/app-api/serverless.yml`.
+
+```yaml
+reports:
+    handler: src/handlers/reports.main
+    events:
+        - http:
+              path: reports
+              method: get
+              cors: true
+              authorizer: aws_iam
+    layers:
+        - !Ref PrismaClientEngineLambdaLayer
+        - arn:aws:lambda:us-east-1:901920570463:layer:aws-otel-nodejs-amd64-ver-1-2-0:1
+    vpc:
+        securityGroupIds:
+            - ${self:custom.sgId}
+        subnetIds: ${self:custom.privateSubnets}
+```
+
+The code that runs when this handler is called is in `/app-api/src/handlers/reports.ts`.
+
+The handler points to `/app-api/src/handlers/reports.main`. We're telling the handler where to look for the file that contains the code that will run in the lambda (`src/handlers`) and also the name of the function (`main`). In `reports.ts` the exported handler is called `main`. The location and exported method of `reports.ts` have to match what's specified on the `handler:` line in our YAML file.
+
+We also provide a `path`. This handler will run when someone navigates to the site's base name, slash reports. For prod, it would be `https://mc-review.onemac.cms.gov/reports`.
+
+The `authorizer` line tells AWS that this method can only be invoked by a user who is logged in and has permissions set in AWS IAM. This will typically flow from someone being a state or CMS user, with permission to interact with the main site.
+
+`layers` tie this handler to services that it needs. In this case, it needs database access, so there's a Prisma layer, and it's also being tied to `otel`, our open telemetry service, for observability.
+
+In order to communicate with those layers, we have to specify which `vpc`, or virtual private cloud, this handler will run within.
```

Comment:
`layers` and `vpc` are still a bit mysterious to me. Is there anyway to flesh this out more? For instance, the need for a `prisma` layer to access the service is straightforward, but where would I go in order to find the specifics for the layers `PrismaClientEngineLambdaLayer` and `arn:aws:lambda:us-east-1:901920570463:layer:aws-otel-nodejs-amd64-ver-1-2-0:1` for `otel`. Same question for `vpc`.

Reply:
These lines contain the sum total of my knowledge of layers and vpcs 😆  @mojotalantikite is there a quick summary you can think to include here?

Reply:
Sure! So VPCs we don't have access to control at CMS, so we have one in each environment that was provisioned for us by them. The identifiers we need are stored in Parameter Store, and we use those to configure the VPC that some of these lambdas need access to. The gist of it is: if a handler needs to talk to the DB, it needs to have the VPC config.

That's a good point about the Prisma client engine layer, we don't really have docs about that. It's a layer that is built from the `app-api/scripts/prepare-prisma-layer.sh` script, which is needed to communicate to the DB.

As for the OTEL layer, that is one that is made by [AWS and we just import.](https://aws-otel.github.io/docs/getting-started/lambda/lambda-js) We can write some docs about it -- there's actually an updated layer that we might want to schedule an upgrade to our OTEL tooling for.